queue/abstract.lua

```diff
@@ -374,6 +374,28 @@ local function make_self(driver, space, tube_name, tube_type, tube_id, opts)
     return self
 end
 
+--- Release all session tasks.
+local function release_session_tasks(connection_id)
+    while true do
+        task = box.space._queue_taken.index.pk:min{connection_id}
```

Comment:
This variable is not declared.

Also I think you could use `ipairs` + REQ iterator here.
Consider following diff:

```diff
diff --git a/queue/abstract.lua b/queue/abstract.lua
index d666751..8a53aa6 100644
--- a/queue/abstract.lua
+++ b/queue/abstract.lua
@@ -376,12 +376,7 @@ end
 
 --- Release all session tasks.
 local function release_session_tasks(connection_id)
-    while true do
-        task = box.space._queue_taken.index.pk:min{connection_id}
-        if task == nil or task[1] ~= connection_id then
-            break
-        end
-
+    for _, task in box.space._queue_taken.index.pk:pairs({connection_id}, {iterator = box.index.REQ}) do
         local tube = box.space._queue.index.tube_id:get{task[2]}
         if tube == nil then
             log.error("Inconsistent queue state: tube %d not found", task[2])
```

I think it should be faster. Main problem here is stability of TREE iterators (https://github.com/tarantool/tarantool/commit/2160e4cbdb90caff0a3c766bd92003d1f7d67d7b), we could break compatibility with old versions. But I think we could workaround it separately (but anyway since we don't have 1.7 in our CI we don't have any guarantees it works just now).

Reply:
> Also I think you could use `ipairs` + REQ iterator here.
> Consider following diff:

https://github.com/tarantool/queue/pull/139#discussion_r544166212

Reply:
> This variable is not declared.

Yes. You are right (I haven't checked intermediate patches with `luachek` and I missed this mistake).